app/src/main/java/com/android/calculator2/widget/CalculatorWidget.java

```diff
@@ -60,6 +61,9 @@ public void onReceive(Context context, Intent intent) {
         if(value.equals(context.getResources().getString(R.string.error_syntax))) value = "";
         mClearText = intent.getBooleanExtra(SHOW_CLEAR, false);
 
+        String action = intent.getAction().substring(intent.getAction().lastIndexOf(".")+1);
```

Comment:
The expected values are the constants defined in CalculatorWidget.java, used by the widget to identify the button clicked.

This is an example:

`public static final String DIGIT_0 = "com.android.calculator2.zero";`